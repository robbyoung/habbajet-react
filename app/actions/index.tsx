export * from './actionTypes';
export * from './addHabbajet';
export * from './addHabitResult';
export * from './updateBudget';
export * from './resetHabbajet';
export * from './loadState';
export * from './selectHabbajet';
export * from './updateEditorField';
export * from './validateEditor';
export * from './clearEditor';
