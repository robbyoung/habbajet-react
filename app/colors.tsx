export const white = '#ffffff';
export const lightGrey = 'F5F5F5';
export const grey = '#959595';

export const habbajetColors = [
    '#8066C2', // Purple
    '#2AAD5E', // Green
    '#E04747', // Red
    '#7F90EC', // Blue
    '#B1B952', // Gold
];

export const complementColors: {[id: string]: string} = {
    '#8066C2': '#C2A066',
    '#2AAD5E': '#AD372A',
    '#E04747': '#47E047',
    '#7F90EC': '#ECB27F',
    '#B1B952': '#B952B0',
};
