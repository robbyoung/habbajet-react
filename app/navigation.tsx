import {Navigation} from 'react-native-navigation';
import React from 'react';
import HabbajetScreen from './containers/habbajetScreen';
import LoadingScreen from './containers/loadingScreen';
import {Provider} from 'react-redux';
import store from './store';

enum Screens {
    Loading = 'Loading',
    Habbajet = 'Habbajet',
}

export const STACK_NAVIGATOR = 'StackNavigator';

Navigation.registerComponent(
    Screens.Loading,
    () => () => (
        <Provider store={store}>
            <LoadingScreen />
        </Provider>
    ),
    () => LoadingScreen,
);

Navigation.registerComponent(
    Screens.Habbajet,
    () => () => (
        <Provider store={store}>
            <HabbajetScreen />
        </Provider>
    ),
    () => HabbajetScreen,
);

export const goToLoading = () => {
    Navigation.setRoot({
        root: {
            stack: {
                id: STACK_NAVIGATOR,
                children: [
                    {
                        component: {
                            name: Screens.Loading,
                            options: {
                                topBar: {
                                    visible: false,
                                },
                            },
                        },
                    },
                ],
            },
        },
    });
};

export const goToHabbajet = () => {
    Navigation.setRoot({
        root: {
            stack: {
                id: STACK_NAVIGATOR,
                children: [
                    {
                        component: {
                            name: Screens.Habbajet,
                            options: {
                                topBar: {
                                    title: {
                                        text: ' ',
                                        fontFamily: 'Abel',
                                        fontSize: 30,
                                        color: '#ffffff',
                                    },
                                },
                            },
                        },
                    },
                ],
            },
        },
    });
};
