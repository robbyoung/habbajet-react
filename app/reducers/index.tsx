import habbajets from './habbajets';
import purchases from './purchases';
import budget from './budget';
import habbajetEditor from './habbajetEditor';
import purchaseEditor from './purchaseEditor';
import tagEditor from './tagEditor';
import tags from './tags';
import {combineReducers} from 'redux';

export default combineReducers({
    habbajets,
    purchases,
    budget,
    tags,
    habbajetEditor,
    purchaseEditor,
    tagEditor,
});
