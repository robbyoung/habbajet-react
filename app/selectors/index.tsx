export * from './budget';
export * from './purchases';
export * from './habbajets';
export * from './habbajetEditor';
