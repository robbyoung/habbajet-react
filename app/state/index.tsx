export enum HabitResult {
    Success,
    SlackSuccess,
    Failure,
}

export interface Habbajet {
    id: string;
    name: string;
    maxValue: number;
    currentValue: number;
    modifier: number;
    results: HabitResult[];
    color: string;
    date: string;
    toClaim: boolean;
    selected: boolean;
    currentStreak: number;
    bestStreak: number;
    oldStreaks: [number, number];
    totalSlack: number;
    remainingSlack: number;
}

export interface Purchase {
    id: string;
    name: string;
    cost: number;
    date: string;
    tagId: string;
}

export interface EditorField {
    value: string;
    errorMessage: string;
}

export interface HabbajetEditor {
    name: EditorField;
    value: EditorField;
    modifier: EditorField;
    slack: EditorField;
    color: string;
    validated: boolean;
}

export interface PurchaseEditor {
    name: EditorField;
    cost: EditorField;
    tagId: string;
    validated: boolean;
    id?: string;
}

export interface Tag {
    id: string;
    name: string;
    color: string;
}

export interface State {
    habbajets: Habbajet[];
    purchases: Purchase[];
    tags: Tag[];
    habbajetEditor: HabbajetEditor;
    purchaseEditor: PurchaseEditor;
    budget: number;
}
