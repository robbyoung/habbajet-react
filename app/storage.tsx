import AsyncStorage from '@react-native-community/async-storage';
import {State} from './state';
import store from './store';
import {createTestState} from './state/testState';

// const defaultState: State = {
//     budget: 0,
//     habbajets: [],
//     purchases: [],
// };

const defaultState = createTestState(1, 0, 0);

export async function saveState(): Promise<void> {
    const state = store.getState();
    await AsyncStorage.setItem('state', JSON.stringify(state));
}

export async function loadState(): Promise<State> {
    const state = await AsyncStorage.getItem('state');
    let savedState: State;
    if (state !== undefined && state !== null) {
        savedState = JSON.parse(state) as State;
        return savedState;
    }
    return defaultState;
}

export async function clearState() {
    await AsyncStorage.clear();
}
