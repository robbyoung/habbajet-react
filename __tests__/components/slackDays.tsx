import React from 'react';
import renderer from 'react-test-renderer';
import {habbajetColors} from '../../app/colors';
import SlackDays from '../../app/components/slackDays';

describe('SlackDays Component', () => {
    it('renders circles to represent slack days', () => {
        const component = renderer.create(
            <SlackDays color={habbajetColors[0]} total={6} remaining={6} />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders grey circles for used slack days', () => {
        const component = renderer.create(
            <SlackDays color={habbajetColors[1]} total={3} remaining={0} />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('hides if there are no slack days', () => {
        const component = renderer.create(
            <SlackDays color={habbajetColors[2]} total={0} remaining={0} />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
