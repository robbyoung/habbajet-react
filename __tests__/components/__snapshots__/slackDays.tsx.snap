// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlackDays Component hides if there are no slack days 1`] = `<View />`;

exports[`SlackDays Component renders circles to represent slack days 1`] = `
<View
  style={
    Object {
      "marginBottom": 10,
      "marginTop": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Abel",
          "fontSize": 20,
          "margin": 0,
          "padding": 0,
        },
        Object {
          "color": "#8066C2",
        },
      ]
    }
  >
    Slack Days
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 15,
            "height": 30,
            "marginRight": 5,
            "width": 30,
          },
          Object {
            "backgroundColor": "#8066C2",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "borderRadius": 15,
            "height": 30,
            "marginRight": 5,
            "width": 30,
          },
          Object {
            "backgroundColor": "#8066C2",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "borderRadius": 15,
            "height": 30,
            "marginRight": 5,
            "width": 30,
          },
          Object {
            "backgroundColor": "#8066C2",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "borderRadius": 15,
            "height": 30,
            "marginRight": 5,
            "width": 30,
          },
          Object {
            "backgroundColor": "#8066C2",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "borderRadius": 15,
            "height": 30,
            "marginRight": 5,
            "width": 30,
          },
          Object {
            "backgroundColor": "#8066C2",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "borderRadius": 15,
            "height": 30,
            "marginRight": 5,
            "width": 30,
          },
          Object {
            "backgroundColor": "#8066C2",
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`SlackDays Component renders grey circles for used slack days 1`] = `
<View
  style={
    Object {
      "marginBottom": 10,
      "marginTop": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Abel",
          "fontSize": 20,
          "margin": 0,
          "padding": 0,
        },
        Object {
          "color": "#E04747",
        },
      ]
    }
  >
    Slack Days
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 15,
            "height": 30,
            "marginRight": 5,
            "width": 30,
          },
          Object {
            "backgroundColor": "#6E6E6E",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "borderRadius": 15,
            "height": 30,
            "marginRight": 5,
            "width": 30,
          },
          Object {
            "backgroundColor": "#6E6E6E",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "borderRadius": 15,
            "height": 30,
            "marginRight": 5,
            "width": 30,
          },
          Object {
            "backgroundColor": "#6E6E6E",
          },
        ]
      }
    />
  </View>
</View>
`;
